### ISSUE TITLE
Metric names should not be modified

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#3275]
There are a few problems caused by manipulation of metric names for generating the verbose name and for displays. Here is an example with 2 reports, they use different metrics whose name only differ due to case

```python
from sklearn.dummy import DummyClassifier
from sklearn.datasets import make_classification
from sklearn.metrics import accuracy_score

import skore

X, y = make_classification()

report = skore.CrossValidationReport(DummyClassifier(), X, y)


def MY_METRIC(e, X, y):
    return accuracy_score(e.predict(X), y)


report.metrics.add(MY_METRIC)

print(report.metrics.summarize().frame())

skore.login()
project = skore.Project(workspace="jerome-workspace-1", name="project-3", mode="hub")
project.put("report", report)


report_1 = skore.CrossValidationReport(DummyClassifier(), X, y)


def My_metric(e, X, y):
    return accuracy_score(e.predict(X), y)


report_1.metrics.add(My_metric)

project.put("report_1", report_1)

print(report_1.metrics.summarize().frame())


comparison = skore.ComparisonReport([report, report_1])
print(comparison.metrics.summarize())
```

in the individual project local display, the metric appears as `my_metric` (case has changed):

```
              dummyclassifier_mean  dummyclassifier_std
my_metric                 0.510000         2.236068e-02
score                     0.510000         2.236068e-02
accuracy                  0.510000         2.236068e-02
precision_0               0.510000         2.236068e-02
precision_1               0.000000         0.000000e+00
recall_0                  1.000000         0.000000e+00
recall_1                  0.000000         0.000000e+00
roc_auc                   0.500000         0.000000e+00
log_loss                  0.693397         1.397979e-04
brier_score               0.250125         6.987712e-05
fit_time                  0.000083         1.627909e-05
predict_time              0.000007         5.100729e-07
```

in the local comparison report, it appears as `my_metric` but they are still considered as 2 separate metrics, with duplicates in the index and NaNs:

```
              mean_dummyclassifier_1  ...  std_dummyclassifier_2
my_metric                   0.510000  ...                    NaN
score                       0.510000  ...           2.236068e-02
accuracy                    0.510000  ...           2.236068e-02
precision_0                 0.510000  ...           2.236068e-02
precision_1                 0.000000  ...           0.000000e+00
recall_0                    1.000000  ...           0.000000e+00
recall_1                    0.000000  ...           0.000000e+00
roc_auc                     0.500000  ...           0.000000e+00
log_loss                    0.693397  ...           1.397979e-04
brier_score                 0.250125  ...           6.987712e-05
fit_time                    0.000090  ...           1.627909e-05
predict_time                0.000008  ...           5.100729e-07
my_metric                        NaN  ...           2.236068e-02
```

in the hub they are both displayed as My Metric which also differs from the actual name, with the same problem of seemingly duplicate name and NaNs

<img width="2044" height="211" alt="Image" src="https://github.com/user-attachments/assets/69c6add2-bd43-4543-bb75-853751280041" />

I think we should not have the name and verbose name, only a name which should be exactly the name provided by the user (either explicitly or the function name). we should not perform any modifications on the name such as changing case or replacing some characters with spaces.

we can keep a verbose name if we want, but then it should just be stored and used for display on the hub. it should not be used in the lib, and if not provided it should fall back to the exact name, without transformations

ping @auguste-probabl who knows the metrics handling best and probably has clearer ideas on this topic

**Comment by glemaitre** [comment]
> we should not perform any modifications on the name such as changing case or replacing some characters with spaces.

I agree.

> it should not be used in the lib

I think we can still use it. For instance, `.summary` in `MetricsSummaryDisplay` is exposing a `verbose_name` parameter. So it is quite explicit and it should not be a problem.

**Comment by m4nn2609-dot** [comment]
I agree with @jeromedockes and @glemaitre on the naming of parameters, I think we should keep the verbose name too.
